src/Context/Traits/UserAwareContextTrait.php

```diff
@@ -84,31 +85,134 @@ public function loggedIn()
         return false;
     }
 
-
-
     /**
      * Create a user.
+     * If the user already exists, the existing user will be
+     * compared with the user which was asked to be created. If all matches
+     * no fault with be thrown. If it does not match then UnexpectedValueException
+     * will be thrown.
+     *
+     * @param string $userLogin User login name.
+     * @param string $userEmail User email address.
+     * @param array  $args      Optional. Extra parameters to pass to WordPress.
      *
-     * @param string $userLogin  User login name.
-     * @param string $userEmail  User email address.
-     * @param array  $args        Optional. Extra parameters to pass to WordPress.
+     * @throws \UnexpectedValueException
      *
      * @return array {
-     *         @type int $id User ID.
-     *         @type string $slug User slug (nicename).
+     *             @type int $id User ID.
+     *             @type string $slug User slug (nicename).
      *         }
      */
     public function createUser($userLogin, $userEmail, $args = [])
     {
         $args['user_email'] = $userEmail;
         $args['user_login'] = $userLogin;
 
-        $user = $this->getDriver()->user->create($args);
+        try {
+            $user = $this->getDriver()->user->create($args);
+        } catch (UnexpectedValueException $exception) {
+            $user = $this->getExistingMatchingUser($args);
+        }
 
-        return array(
+        $return_array = array(
             'id'   => $user->ID,
             'slug' => $user->user_nicename
         );
+
+        return $return_array;
+    }
+
+    /**
+     * Returns a user if all the passed parameters match
```

Comment:
This reads like it should be the description for the `return` phpdoc, not the function itself.

Reply:
agree. changed